contracts/amm/contract/src/lib.rs

```diff
@@ -73,11 +145,11 @@ impl AmmState {
 }
 
 impl AmmContract {
-    pub fn new(contract_name: ContractName, state: AmmState, caller: Identity) -> Self {
+    pub fn new(exec_state: ExecutionState, contract_name: ContractName, state: AmmState) -> Self {
```

Comment:
the naming of ExecutionState confuses me. I have 2 states to manage in my contract ? I would better understand the function if I have a Context & a State